### PR TITLE
pd/heartbeat: report the number of rows in region

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,
      This PR will report the approximate number of rows in the region during the heartbeat.
It's prepared for controlling the region by both the number of rows and size.

@huachaohuang @overvenus PTAL

### CODE REVIEW

File: src/pd/client.rs [hunk 1]
```diff
@@ -299,7 +299,8 @@ impl PdClient for RpcClient {
         req.set_keys_written(region_stat.written_keys);
         req.set_bytes_read(region_stat.read_bytes);
         req.set_keys_read(region_stat.read_bytes);
-        req.set_approximate_size(region_stat.approximate_size);
+        req.set_approximate_size(region_stat.approximate_stat.size);
+        req.set_approximate_count(region_stat.approximate_stat.rows);
```
Comment (overvenus) [inline src/pd/client.rs]: Please update to the latest version.

File: src/pd/pd.rs [hunk 2]
```diff
@@ -540,11 +540,11 @@ impl<T: PdClient> Runnable<Task> for Runner<T> {
...
                 written_keys,
-                region_size,
+                approximate_stat,
             } => {
-                let approximate_size = match region_size {
-                    Some(size) => size,
-                    None => get_region_approximate_size(&self.db, &region).unwrap_or(0),
+                let approximate_stat = match approximate_stat {
+                    Some(stat) => stat,
+                    None => RegionApproximateStat::new(&self.db, &region).unwrap_or_default(), //get_region_approximate_size(&self.db, &region).unwrap_or(0),
```
Comment (overvenus) [inline src/pd/pd.rs]: Please clean the comment.

File: src/import/engine.rs [hunk 3]
```diff
@@ -122,6 +122,8 @@ fn tune_dboptions_for_bulk_load(opts: &DbConfig) -> (DBOptions, CFOptions) {
     // Add size properties to get approximate ranges wihout scan.
     let f = Box::new(SizePropertiesCollectorFactory::default());
     cf_opts.add_table_properties_collector_factory("tikv.size-properties-collector", f);
+    let f = Box::new(MvccPropertiesCollectorFactory::default());
```
Comment (zhangjinpeng1987) [inline src/import/engine.rs]: Why default cf need this?

File: src/raftstore/store/msg.rs [hunk 4]
```diff
@@ -162,10 +163,10 @@ pub enum Msg {
         hash: Vec<u8>,
     },
 
-    // For region size
-    ApproximateRegionSize {
+    // For region stat
+    ApproximateRegionStat {
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/msg.rs]: ApproximateRegionStat and RegionApproximateStat, oh, they will cause confusing.

Comment (huachaohuang) [reply src/raftstore/store/msg.rs]: Can we use `RegionApproximateStat` here too? Will it conflict?

**Pull request merged**